docs/src/content/hardhat-runner/docs/guides/test-contracts.md

```diff
@@ -1,3 +1,197 @@
+
+While this is the way we recommend, Hardhat is flexible and you can customize the approach or take a completely different path with other tools.
+
+### Initial setup
+
+In this guide we’ll write some tests for the sample project so, if you haven’t done it yet, go and [initialize it](./project-setup.md).
+
+We also recommend you use [TypeScript](./typescript.md) to get better autocompletion and catch possible errors earlier. This guide will assume you are using TypeScript, but you can click the tabs of the snippets to switch their language.
+
+The setup includes some example tests in the `test/Lock.ts` file, but ignore them for now. Instead, create a `test/my-tests.ts` file directory and run only those. For example, if your test file is called `test/my-tests.ts`, you can run only that file with `npx hardhat test test/my-tests.ts`
+
+### A simple test
+
+For our first test we’ll deploy the `Lock` contract and assert that the unlock time returned by the `unlockTime()` getter is the same one that we passed in the constructor:
+
+```tsx
+import { expect } = from "chai";
+import hre from "hardhat";
+import helpers from "hardhat-network-helpers";
+
+describe("Lock", function () {
+  it("Should set the right unlockTime", async function () {
+    const lockedAmount = 1_000_000_000;
+    const ONE_YEAR_IN_SECS = 365 * 24 * 60 * 60;
+    const unlockTime = (await helpers.time.latest()) + ONE_YEAR_IN_SECS;
+
+    // deploy a lock contract where funds can be withdrawn
+    // one year in the future
+    const Lock = await ethers.getContractFactory("Lock");
+    const lock = await Lock.deploy(unlockTime, { value: lockedAmount });
+
+    // assert that the value is correct
+    expect(await lock.unlockTime()).to.equal(unlockTime);
+  });
+});
+```
+
+The first thing we do is to import the functions we are going to use: the [`expect`](https://www.chaijs.com/api/bdd/) function from `chai` to write our assertions, the [Hardhat Runtime Environment](../advanced/hardhat-runtime-environment.md), and the [network helpers](/hardhat-network-helpers) to interact with the Hardhat Network. After that we use the `describe` and `it` functions, which are global Mocha's functions used to describe and group your tests (you can read more about Mocha [here](https://mochajs.org/#getting-started)).
+
+The test itself is what’s inside the `it` function. First we set the values for the amount we want to lock (in [wei](https://ethereum.org/en/glossary/#wei)) and the unlock time. For the latter we use [`time.latest`](</hardhat-network-helpers/docs/reference#latest()>), a network helper that returns the timestamp of the last mined block. Then we deploy the contract itself: first we get a [`ContractFactory`](https://docs.ethers.io/v5/single-page/#/v5/api/contract/contract-factory/) for the `Lock` contract and then we deploy it, passing the unlock time as its only constructor argument.
```

Comment:
We also pass it a value

Reply:
0f82fdf3a0a9c51f3faa54ebba1a3afef4ebb04f

Reply:
I'd be more explicit. Something like "We also pass it an object with transaction (not constructor) parameters. This is optional, but will use it to send some ETH by setting its `value` field."

Reply:
Added something like that in 834aa5eac25661b1d40caf699e2442e7aef31978